AppCenterDataStorage/AppCenterDataStorageTests/MSTokenExchangeTests.m

```diff
@@ -218,6 +217,48 @@ - (void)testValidCachedTokenExists {
   [utilityMock stopMocking];
 }
 
+- (void)testRetrieveCachedTokenFails {
```

Comment:
Can you update the test method name or add a comment what you are testing here?
The test name says it is testing a failure case when it retrieves cached token but there is a HTTP client mock that also returns failed token. The test contains two failure tokens 1. in cache 2. from HTTP response. What are you testing here?